extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -697,37 +667,61 @@ public String getErrorMessage() {
     private static final String HTTPS = "https:";
     private static final String DEOBFUSCATION_FUNC_NAME = "deobfuscate";
 
-    private final static String[] REGEXES = {
+    private static final String[] REGEXES = {
             "(?:\\b|[^a-zA-Z0-9$])([a-zA-Z0-9$]{2})\\s*=\\s*function\\(\\s*a\\s*\\)\\s*\\{\\s*a\\s*=\\s*a\\.split\\(\\s*\"\"\\s*\\)",
             "([\\w$]+)\\s*=\\s*function\\((\\w+)\\)\\{\\s*\\2=\\s*\\2\\.split\\(\"\"\\)\\s*;",
             "\\b([\\w$]{2})\\s*=\\s*function\\((\\w+)\\)\\{\\s*\\2=\\s*\\2\\.split\\(\"\"\\)\\s*;",
-            "yt\\.akamaized\\.net/\\)\\s*\\|\\|\\s*.*?\\s*c\\s*&&\\s*d\\.set\\([^,]+\\s*,\\s*(:encodeURIComponent\\s*\\()([a-zA-Z0-9$]+)\\(",
             "\\bc\\s*&&\\s*d\\.set\\([^,]+\\s*,\\s*(:encodeURIComponent\\s*\\()([a-zA-Z0-9$]+)\\("
     };
+    private static final String STS_REGEX = "signatureTimestamp[:=]\\\\d+";
```

Comment:
This regular expression doesn't work, no matches are found, contrary to the string of the original JavaScript regular expression ([that I found here](https://github.com/Rafostar/clapper/blob/fe03719b38b2e925462d8ff02488fde7abe96abf/src/youtube.js#L222)). Can you help me please to find a working regular expression?

Reply:
Does `"signatureTimestamp[=\\:]\\d+"` work?

I basically followed https://stackoverflow.com/a/57575475

Reply:
Unfortunately no, it doesn't work.

Reply:
This is not a correct regex. `\\\\` should be replaced with `\\`, as the earlier matches only a literal backslash in the string.

Reply:
Thank you, it works! The real issue was that I used this regular expression to parse the embed HTML code, not the JS player code.

Reply:
Great! However, I don't think it would work with the four backslashes; did you change it locally?

> Edit: However, it includes signatureTimestamp:: I got signatureTimestamp:18772 from my tests. Are there better methods than removing the non-digits chars?

Try surrounding the `\\d+` with parenthesis, like this: `signatureTimestamp[:=](\\d+)`.

Reply:
It's the same result. As a workaround, I use `Utils.removeNonDigitCharacters(Parser.matchGroup(STS_REGEX, playerCode, 0))`.

Reply:
Either pass `0` instead of `1` (group 0 is the matched string), or use `matchGroup1` instead of `matchGroup`.